### PR TITLE
FC loss - clip focal_weight & set reduction to AUTO

### PR DESCRIPTION
**Pull request by johnypark**
# Description

Brief Description of the PR: 

The current focal loss has a bug that induces a complex number when gamma <1, which may be due to leakage in floating number calculation that results in negative numbers in probability. 

Also, with the default reduction =NONE,  exploding gradient case is observed with an adequate learning rate.

Made the simple following changes:
- Use the keras backend clip function to set minimum focal_weight values with epsilon, which solves the negative number problem.
-  Set reduction = AUTO, which mitigates the exploding gradient problem. 

## Type of change

- [c ] Bug fix

# How Has This Been Tested?

Tested it on a [Colab notebook](https://colab.research.google.com/drive/1tp4e-WtN39HjL9jS7aPulE0Bp_U1kFtX?usp=sharing) with TPU and tensorflow=2.8 and tensorflow_addons=0.16.1.

Please let me know if we need a more exhaustive demonstration, I can provide some more tests on Colab. Allowing edits by maintainers just in case.

### CODE REVIEW

**Comment by bhack** [comment]
Do you have a test to cover this case?

**Comment by bhack** [comment]
Do you have a test to cover this case?

**Comment by johnypark** [comment]
> Do you have a test to cover this case?

Hi,  I did the test here [Colab notebook](https://colab.research.google.com/drive/1tp4e-WtN39HjL9jS7aPulE0Bp_U1kFtX?usp=sharing) ... the class fixed_SigmoidFocalCrossEntropy includes the two simple changes made to the code and the notebook shows that it works. 

I am new to contributing to tensorflow addons, so not familiar of what is the standard way of doing the test. Please direct me what kinds of test needs to be done to cover this case.

**Comment by AakashKumarNain** [comment]
@bhack what's the plan of incorporating Focal loss from Keras-cv?

**Comment by bhack** [comment]
> @bhack what's the plan of incorporating Focal loss from Keras-cv?
> 

We have a contribuor interested https://github.com/keras-team/keras-cv/issues/319#issuecomment-1104592455

**Comment by AakashKumarNain** [comment]
> > @bhack what's the plan of incorporating Focal loss from Keras-cv?
> 
> We have a contribuor interested [keras-team/keras-cv#319 (comment)](https://github.com/keras-team/keras-cv/issues/319#issuecomment-1104592455)

Let's try to get that done and move this one away slowly

**Comment by johnypark** [comment]
> > > @bhack what's the plan of incorporating Focal loss from Keras-cv?
> > 
> > 
> > We have a contribuor interested [keras-team/keras-cv#319 (comment)](https://github.com/keras-team/keras-cv/issues/319#issuecomment-1104592455)
> 
> Let's try to get that done and move this one away slowly

Can you review and incorporate this changes to addons if you like them, in the meantime? They are just simple two changes in the code.

**Comment by bhack** [comment]
Can you add a test for this uncovered case in https://github.com/tensorflow/addons/blob/master/tensorflow_addons/losses/tests/focal_loss_test.py
